src/main/java/com/linkedin/kmf/services/ConsumerFactoryImpl.java

```diff
@@ -81,10 +81,27 @@ public ConsumerFactoryImpl(Map<String, Object> props) throws Exception {
       props.forEach(consumerProps::putIfAbsent);
     }
 
-    _baseConsumer = (KMBaseConsumer) Class.forName(consumerClassName)
-        .getConstructor(String.class, Properties.class, AdminClient.class)
-        .newInstance(_topic, consumerProps, adminClient());
+    if (adminClientConstructorExists(consumerClassName)) {
+      _baseConsumer = (KMBaseConsumer) Class.forName(consumerClassName)
+          .getConstructor(String.class, Properties.class, AdminClient.class)
+          .newInstance(_topic, consumerProps, adminClient());
+    } else {
+      _baseConsumer = (KMBaseConsumer) Class.forName(consumerClassName)
+          .getConstructor(String.class, Properties.class)
+          .newInstance(_topic, consumerProps);
+    }
+  }
 
+  static boolean adminClientConstructorExists(String consumerClassName) throws ClassNotFoundException {
+    try {
+      Class.forName(consumerClassName).getConstructor(String.class, Properties.class, AdminClient.class);
+      return true;
+    } catch (java.lang.NoSuchMethodException noSuchMethodException) {
+      LOG.error("The method was not found. This is not a constructor with AdminClient parameter: ", noSuchMethodException);
```

Comment:
I think this should be an info or debug log. i.e., it is not an error since you fall back to the properties-based constructor.